executor/hnswpsql.py

```diff
@@ -352,25 +350,23 @@ def _start_auto_sync(self):
         self.sync_thread.start()
 
     def close(self) -> None:
-        global STOP_SYNC_THREAD
-
         if hasattr(self, 'sync_thread'):
-            STOP_SYNC_THREAD = True
+            # wait for sync thread to finish
+            self.stop_sync_thread = True
             try:
                 while self.sync_thread.is_alive():
                     time.sleep(2)
```

Comment:
simply use join

Reply:
didn't you say not to use join and use a flag?

Reply:
u use join but not with timeout. U use join to wait for the flag to take its effects. The problem was using join with a timeout setting

Reply:
ah I see. Fixed